apps/server/src/modules/oauth/service/iserv-oauth.service.spec.ts

```diff
@@ -0,0 +1,96 @@
+import { Collection } from '@mikro-orm/core';
+import { HttpModule } from '@nestjs/axios';
+import { NotFoundException } from '@nestjs/common';
+import { Test, TestingModule } from '@nestjs/testing';
+import { Role, School, System, User } from '@shared/domain';
+import { UserRepo } from '@shared/repo/user/user.repo';
+import { ObjectId } from 'bson';
+import { Logger } from '@src/core/logger';
+import { createMock } from '@golevelup/ts-jest';
+import { OAuthSSOError } from '../error/oauth-sso.error';
+import { IservOAuthService } from './iserv-oauth.service';
+
+describe('IservOAuthService', () => {
+	let service: IservOAuthService;
+	let userRepo: UserRepo;
+
+	const defaultUserId = '123456789';
+	const defaultSystemId = '987654321';
+	const defaultScool: School = {
+		name: '',
+		_id: new ObjectId(),
+		id: '',
+		systems: new Collection<System>([]),
+	};
+	const defaultUser: User = {
+		email: '',
+		roles: new Collection<Role>([]),
+		school: defaultScool,
+		_id: new ObjectId(),
+		id: '',
+		createdAt: new Date(),
+		updatedAt: new Date(),
+		ldapId: '1111',
+		firstName: '',
+		lastName: '',
+	};
+
+	beforeEach(async () => {
+		const module: TestingModule = await Test.createTestingModule({
+			imports: [HttpModule],
+			providers: [
+				IservOAuthService,
+				{
+					provide: Logger,
+					useValue: createMock<Logger>(),
+				},
+				{
+					provide: UserRepo,
+					useValue: {
+						findByLdapId(userId, systemId) {
```

Comment:
you can mock the return value in the specific test/setup where you need a specific response. No need for logic up here that I need to keep in the back of my head to understand the tests.

Reply:
Actually it is a quite simple "logic" solution for: if u get parameters, return the value, if you don't get parameters, return an exception. We have learned, that everything we can mock in the setup we don't have to think about later on. This also helps us in our every day struggle with the linter ;) (This is a sub 100 lines file and the mock is for 2 out of 4 tests. It is a completed implementation with no more development efforts planned in the future. I think the setup logic is manageable)